File: Dataset/Leetcode/train/66/436.java
class Solution {
    public int[] XXX(int[] digits) {
        int n = digits.length - 1;
        while (n >= 0) {
            if (digits[n] != 9) {
                digits[n]++;
                return digits;
            } else {
                digits[n] = 0;
                n = n - 1;
            }
        }
        int[] ints = new int[digits.length + 1];
        ints[0] = 1;
        return ints;
    }
}

File: Dataset/Leetcode/valid/62/436.cpp
class Solution {
public:
    int XXX(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n));
        for (int i = 0; i < m; i ++) {
            for (int j = 0; j < n; j ++) {
                dp[i][j] = i && j ? dp[i][j-1] + dp[i-1][j] : 1;
            }
        }   
        return dp[m-1][n-1];
    }
};

